db/src/tx.rs

```diff
@@ -685,32 +689,56 @@ impl<'conn, 'a> Tx<'conn, 'a> {
     }
 }
 
-/// Transact the given `entities` against the given SQLite `conn`, using the given metadata.
-/// If you want this work to occur inside a SQLite transaction, establish one on the connection
-/// prior to calling this function.
-///
-/// This approach is explained in https://github.com/mozilla/mentat/wiki/Transacting.
-// TODO: move this to the transactor layer.
-pub fn transact<'conn, 'a, I>(
-    conn: &'conn rusqlite::Connection,
-    mut partition_map: PartitionMap,
-    schema_for_mutation: &'a Schema,
-    schema: &'a Schema,
-    entities: I) -> Result<(TxReport, PartitionMap, Option<Schema>)> where I: IntoIterator<Item=Entity> {
-
+/// Initialize a new Tx object with a new tx id and a tx instant. Kick off the SQLite conn, too.
+fn start_tx<'conn, 'a>(conn: &'conn rusqlite::Connection,
+                       mut partition_map: PartitionMap,
+                       schema_for_mutation: &'a Schema,
+                       schema: &'a Schema) -> Result<Tx<'conn, 'a>> {
     let tx_instant = ::now(); // Label the transaction with the timestamp when we first see it: leading edge.
     let tx_id = partition_map.allocate_entid(":db.part/tx");
 
     conn.begin_tx_application()?;
 
-    let mut tx = Tx::new(conn, partition_map, schema_for_mutation, schema, tx_id, tx_instant);
-
-    let report = tx.transact_entities(entities)?;
+    Ok(Tx::new(conn, partition_map, schema_for_mutation, schema, tx_id, tx_instant))
+}
 
+fn conclude_tx(tx: Tx, report: TxReport) -> Result<(TxReport, PartitionMap, Option<Schema>)> {
```

Comment:
`report` is passed through here.  Should it be part of this API?   Even worse, `conclude_tx` has no tie to the underlying `conn`.  Why not just inline this?

Reply:
This is simply extracting the shared boilerplate between `transact` and `transact_terms`, just as we have `start_tx`.